lib/xfrm/sp.c

```diff
@@ -1338,11 +1338,25 @@ int xfrmnl_sp_set_userpolicy_type (struct xfrmnl_sp* sp, unsigned int type)
 	return 0;
 }
 
-void xfrmnl_sp_add_usertemplate(struct xfrmnl_sp *sp, struct xfrmnl_user_tmpl *utmpl)
+/**
+ * @brief      Add a XFRM user template to a XFRM policy.
+ *
+ * @param      sp       policy to add a XFRM template to
+ * @param      len      template to add to the policy
+ *
+ * @return     0 if sucessful, -1 if the number of template of the policy
+ *             would exceed the number of templates allowed for a policy (6)
+ */
+int xfrmnl_sp_add_usertemplate(struct xfrmnl_sp *sp, struct xfrmnl_user_tmpl *utmpl)
 {
-	nl_list_add_tail(&utmpl->utmpl_list, &sp->usertmpl_list);
-	sp->nr_user_tmpl++;
-	sp->ce_mask |= XFRM_SP_ATTR_TMPL;
+	if (MAX_TMPLS > sp->nr_user_tmpl) {
+		nl_list_add_tail(&utmpl->utmpl_list, &sp->usertmpl_list);
+		sp->nr_user_tmpl++;
+		sp->ce_mask |= XFRM_SP_ATTR_TMPL;
+		return 0;
+	}
```

Comment:
An existing function cannot change API in this way (e.g. change a `void` return to `int`). 

Also, is this useful? OK, the user surely doesn't want to configure more than 6 templates (as kernel won't reject it and it doesn't work). But currently the user can set that up in libnl and get a failure later. Now, they will get a failure during `xfrmnl_sp_add_usertemplate`. Does that solve any problem? It won't work for the user either way, and they are well advised to count to 6 themselves.

What if kernel tomorrow supports more than 6 templates?

Just let user-space try to configure the templates they want. If kernel thinks there are too many, it rejects it. That is fine behavior.